### ISSUE TITLE
usb tethering + scrcpy no console= scrcpy closes automatically and never able to launch again

### ISSUE THREAD

**Issue by Vijay-Apple-Dev** [Genymobile/scrcpy#2270]
usb tethering + scrcpy no console= scrcpy closes automatically and never able to launch again.
I hope it was working fine in 1.16, but facing this issue in 1.17
Somebody please enable this feature? thanks a lot :)

**Comment by rom1v** [comment]
Launch `scrcpy-console.bat` to see the error.

**Comment by Vijay-Apple-Dev** [comment]
@rom1v thank you for your quick help..here is the error message...

after enabled USB tethering, it closed automatically.

then relaunched the scrcpy. it says like this.


INFO: scrcpy 1.17 <https://github.com/Genymobile/scrcpy>
adb: error: failed to get feature set: no devices/emulators found
ERROR: "adb push" returned with value 1

**Comment by rom1v** [comment]
Your device is not detected by adb.

This may happen because when you enable USB tethering, the USB devideID may change, and your driver may not recognize/authorize it.

On Linux this can be fixed via new udev rules. On Windows, I don't know. Did you install the latest drivers: https://github.com/Genymobile/scrcpy/blob/master/FAQ.md#device-not-detected

**Comment by Vijay-Apple-Dev** [comment]
@rom1v Yes, I have updated all the drivers to the latest in my windows. any idea to fix this issue!

**Comment by SamuelDasaolu** [comment]
Vijay, I hope it's not too late but to fix that problem next time, run 'adb disconnect' , then rerun scrcpy

**Comment by Vijay-Apple-Dev** [comment]
@SamuelDasaolu so kind of you :) It won't work. I hope we can not use both at the same time "USB tethering for internet to PC + Scrcpy in the same computer using USB".. But it would be nice if we can use both of them..please update me if you have any questions :)

**Comment by SamuelDasaolu** [comment]
It does work for me, I think it's because I enabled  adb in charge only mode from developer options. Try that and see if it works for you also

**Comment by Vijay-Apple-Dev** [comment]
@SamuelDasaolu Thank you so much for your time. But my developer option has no option as "Allow ADB Debugging in charge only" option as in the below url. I'm trying to find out... 

https://consumer.huawei.com/eg-en/support/content/en-us00407281/

**Comment by SamuelDasaolu** [comment]
Then it's probably due to your device model. It's should be a situation peculiar to you. But does scrcpy work on your device while in file transfer mode. If yes, you would just have to make do with that. But to answer your original question whether scrcpy works with USB tethering. It does
